### PR TITLE
Add Device Admission

### PR DESCRIPTION
**Pull request by cl2017**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add admission validation rules for the `device CR` to ensure that the `property.name` is unique.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by WillardHu** [comment]
You need to add the webhook configuration to the ValidatingWebhookConfiguration in the method `registerWebhooks(..)`

**Comment by cl2017** [comment]
> You need to add the webhook configuration to the ValidatingWebhookConfiguration in the method `registerWebhooks(..)`

ok, done.

File: cloud/pkg/admissioncontroller/admit_device.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	//device properties name must be unique.
+	var msg string
+	propertyNameMap := make(map[string]bool)
+	for _, property := range device.Spec.Properties {
+		if _, ok := propertyNameMap[property.Name]; !ok {
+			propertyNameMap[property.Name] = true
+		} else {
+			msg = "property names must be unique."
+			response.Allowed = false
+		}
```
Comment (WillardHu) [inline cloud/pkg/admissioncontroller/admit_device.go:53]: There is no need to use map to determine whether the name is the same
```golang
size := len(device.Spec.Properties)
for i := range device.Spec.Properties {
    for j := i+1; j < size; j++ {
        if device.Spec.Properties[i].Name == device.Spec.Properties[j].Name {
            msg = "property names must be unique."
            response.Allowed = false
            break
        }
    }
}
```

Comment (cl2017) [reply cloud/pkg/admissioncontroller/admit_device.go:53]: I think the time complexity of using a hash table is O(n), which should be lower than that of a two-level loop.

Comment (WillardHu) [reply cloud/pkg/admissioncontroller/admit_device.go:53]: > I think the time complexity of using a hash table is O(n), which should be lower than that of a two-level loop.

Using map requires memory to be allocated and hash code to be calculated. 

When the amount of data is large, the hash code used as map index performs better than loop data.

The value of map doesn't mean much here either.

Comment (cl2017) [reply cloud/pkg/admissioncontroller/admit_device.go]: I have modified

**Comment by wbc6080** [comment]
LGTM

**Pull request merged**